translator.py

```diff
@@ -169,23 +179,32 @@ def _translation_loop(self):
                         # No multicast address currently allocated for this client. Allocate one.
                         mcast_addr = self._alloc_mcast_addr(src_addr)
                         if mcast_addr is not None:
-                            # Publish the new stream on the Multicast Menu if we successfully allocated a multicast
-                            # address. We'll let a worker thread handle the communication with the multicast menu
-                            # s.t. the I/O does not block translation of any ongoing streams (i.e., it is important we
-                            # do not block the thread that runs the translation loop).
+                            # Publish the new stream on Multicast Menu if we successfully allocated a multicast
+                            # address and have a UID to communcate with Multicast Menu. We'll let a worker thread 
+                            # handle the communication with Multicast Menu s.t. the I/O does not block translation 
+                            # of any ongoing streams (i.e., it is important we do not block the thread that runs 
+                            # the translation loop).
                             desc = f'Translated stream originating from {src_addr[0]}'
-                            self._mcastmenu_threadpool.submit(self._add_to_multicast_menu, mcast_addr,
-                                                              constants.MULTICASTMENU_EMAIL, desc)
+                            if self._mm_uid is not None:
+                                self._mcastmenu_threadpool.submit(self._add_to_multicast_menu, mcast_addr, src_addr)
                     if mcast_addr is not None:
                         # Forward the payload to the multicast address allocated for this client.
                         self._mcast_sckt.sendto(payload, (str(mcast_addr), self._mcast_dst_port))
+                        # Update Activity Information Base
+                        self._atb[src_addr] = datetime.now()
                     else:
                         # mcast_addr will be None if we've run out of addresses.
                         # TODO silently discard the packet or communicate this to the client?
                         print(f'WARNING: run out of multicast addresses, cannot serve {src_addr}')
                 except socket.timeout:
                     # No data available to read during this iteration.
                     print('read timeout: nothing to be translated this iteration')
+                # Determine if any streams are newly inactive
+                if self._loop_counter == 30:
+                    self._mcastmenu_threadpool.submit(self._identify_inactive_streams)
```

Comment:
Add todo comment here: `# TODO move this work to a different thread(pool) dedicated to handling clean up`

`_mcastmenu_threadpool` is for sending network requests to the MM. Not much difference from a functional perspective (so no need to change this now), but just to keep things clean.